### PR TITLE
Added ability to push a single image to multiple repositories

### PR DESCRIPTION
**Pull request by fearphage**
Also added an advanced usage readme.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -43,6 +43,10 @@ inputs:
     description: Port the server listens on
     required: false
     default: "3000"
+  registries:
+    description: Credentials adn host for pushing to multiple registries at once
```
Comment (cwebster2) [inline action.yml:47]: s/adn/and/

**Review by cwebster2 — APPROVED** [review]
Looks good to me

File: action.yml [hunk 2]
```diff
@@ -187,6 +191,53 @@ runs:
...
+
+          aws_url='aws://([^:]+):([^@]+)@([0-9a-zA-Z.-]+)'
+          tags=""
+
+          echo "::debug::Found ${#urls[@]} registries entries"
+
+          for url in "${urls[@]}"; do
+            if [[ $url =~ $aws_url ]]; then
+              region="$(cut --fields 4 --delimiter '.' <<< "${BASH_REMATCH[3]}")"
+              echo "::debug::region? $region / #1 ${BASH_REMATCH[1]} / #2 ${BASH_REMATCH[2]} / #3 ${BASH_REMATCH[3]}"
```
Comment (cwebster2) [inline action.yml:216]: is this going to leak secrets or does gh jus turn the key and secret into *********

Comment (fearphage) [reply action.yml:216]: It's not possible. It all comes out as asterisks, but I should remove this debug.

File: advanced_usage.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+```yaml
+- uses: glg-public/build-and-deploy-ecr@main
+  with:
+    access_key_id: ${{ secrets.username }}
+    secret_access_key: ${{ secrets.password }}
+    ecr_uri: ${{ secrets.registry-uri }}
+    registries: |
+      aws://${{ secrets.AWS_ACCESS_KEY_ID }}:${{ secrets.AWS_SECRET_ACCESS_KEY }}@${{ secrets.ECR_URI }}
+      aws://${{ secrets.ANOTHER_KEY_ID }}:${{ secrets.ANOTHER_ACCESS_KEY }}@${{ secrets.ANOTER_ECR_URI }}
```
Comment (shawnrushefsky) [inline advanced_usage.md:53]:
```suggestion
      aws://${{ secrets.ANOTHER_KEY_ID }}:${{ secrets.ANOTHER_ACCESS_KEY }}@${{ secrets.ANOTHER_ECR_URI }}
```

File: advanced_usage.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+a single line of comma-separated values.
+
+
+```yaml
+with:
+  registries: aws://foo:bar@baz,aws://qux:quux@catpants
+```
+
+Note: The `registries` setting is used in conjunction with your default credentials for your main
+build. They are not a replacement for those settings.
```
Comment (shawnrushefsky) [inline advanced_usage.md:66]: Does this mean that it will always build in the default registry, and *also* the list you provide to the registries input?

Comment (fearphage) [reply advanced_usage.md:66]: That is correct. It's not a replacement yet. I didn't have a clean way to pull that off.

Comment (fearphage) [reply advanced_usage.md:66]: It uses [info from the original environment setup](https://github.com/glg-public/build-and-deploy-ecr/pull/21/files#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6R224) to know where to store the additional images.

Comment (fearphage) [reply advanced_usage.md:66]: Also I just tested and you can have duplicate `--tag`s with no problem.

**Review by shawnrushefsky — APPROVED** [review]
